app/controllers/charges_controller.rb

```diff
@@ -9,25 +9,36 @@ def new
   end
 
   def create
+    # I know this controller action is fat. It will be refactored in a future PR
     order = current_user.orders.pending.last
     if order
       # Amount in cents
       amount = order.stripe_total
-      customer = Stripe::Customer.create(
-        email:  current_user.email,
-        source: params[:stripeToken]
-      )
+
+      if session[:stripeToken]
+        customer = Stripe::Customer.create(
```

Comment:
From my understanding, the first time through, a customer is created on Stripe with their credit card info. Stripe generates a token and returns that to my app. I store that token in the user database, so in the future, if that customer wants to use their saved credit card info, they check a box, and I send the saved token to Stripe (instead of creating a new customer), which then creates the charge on the saved card.